### ISSUE TITLE
Zero lines output on *_a3m_wo_ss.ffdata after ffindex_apply_mpi

### ISSUE THREAD

**Issue by tamuanand** [soedinglab/hh-suite#97]
Hi 

I am using HHSuite 3 and I am trying to build a custom database.

When I run the ffindex_apply_mpi step, the *_a3m_wo_ss.ffdata file comes back with 0 lines - is this the expected output? 

Here are the files and the number of lines in each file after ffindex_from_fasta and ffindex_apply_mpi
        0 test_a3m_wo_ss.ffdata
   135273 test_a3m_wo_ss.ffindex
  1093453 test_fas.ffdata
   135273 test_fas.ffindex
  1093453 test.fasta

Command used:
mpirun -np 256 ffindex_apply_mpi test_fas.ff{data,index} -i test_a3m_wo_ss.ffindex -d test_a3m_wo_ss.ffdata -- hhblits -d <path_to>/uniprot20_2016_02 -i stdin -oa3m stdout -n 3 -cpu 1 -v 0

Thanks,
Anand

**Comment by tamuanand** [comment]
Hi Milot,

I created a smaller file with 14 fasta records and ran the program as suggested by you.

There is no error reported - the program runs fine but the _test_a3m_wo_ss.ffdata file comes back empty_.

_This was output produced on stdout_

tr|K0JBI2|K0JBI2_ALOMY  1753    855     18      0
tr|Q1KTF1|Q1KTF1_ZYMTR  0       582     15      0
tr|I2FL51|I2FL51_ALOMY  4598    609     20      0
tr|Q2PDK3|Q2PDK3_ALOMY  2608    750     12      0
tr|A0A0B4LXR3|A0A0B4LXR3_ALOMY  4268    330     423     0
tr|A0A0B4LYA3|A0A0B4LYA3_ALOMY  6025    383     418     0
tr|B2L2R7|B2L2R7_ZYMTR  582     491     396     0
tr|A0A0H4LYV5|A0A0H4LYV5_ALOMY  5207    363     410     0
tr|A0A1B1H4M5|A0A1B1H4M5_ZYMTRtr|B5QSK0|B5QSK0_ALOMY    6408    2469    391     0
        1073    680     402     0
tr|H9AGB9|H9AGB9_ALOMY  3950    318     385     0
tr|B5QSK1|B5QSK1_ALOMY  8877    2469    389     0
tr|H9AQG0|H9AQG0_ALOMY  5570    455     383     0
tr|I2FL21|I2FL21_ALOMY  3358    592     381     0

This is output of _test_a3m_wo_ss.ffindex_

tr|A0A0B4LXR3|A0A0B4LXR3_ALOMY  13      1
tr|A0A0B4LYA3|A0A0B4LYA3_ALOMY  0       1
tr|A0A0H4LYV5|A0A0H4LYV5_ALOMY  1       1
tr|A0A1B1H4M5|A0A1B1H4M5_ZYMTR  2       1
tr|B2L2R7|B2L2R7_ZYMTR  6       1
tr|B5QSK0|B5QSK0_ALOMY  5       1
tr|B5QSK1|B5QSK1_ALOMY  3       1
tr|H9AGB9|H9AGB9_ALOMY  8       1
tr|H9AQG0|H9AQG0_ALOMY  10      1
tr|I2FL21|I2FL21_ALOMY  4       1
tr|I2FL51|I2FL51_ALOMY  11      1
tr|K0JBI2|K0JBI2_ALOMY  7       1
tr|Q1KTF1|Q1KTF1_ZYMTR  9       1
tr|Q2PDK3|Q2PDK3_ALOMY  12      1

**Comment by milot-mirdita** [comment]
Could you try the `hhblits_mpi` binary, that should be also a bit faster and, if it doesn't, work hopefully give at least some reason why.

Something similar to:
```
mpirun -np 256 hhblits_mpi -d <path_to>/uniprot20_2016_02 -i test_fas -oa3m test_a3m_wo_ss -n 3
```

**Comment by tamuanand** [comment]
I got this error message

`mpirun noticed that process rank 0 with PID 22084 on node  exited on signal 4 (Illegal instruction).`

Just checking - does the above command line need to have 

`-i test_fas.ff{data,index}`

**Comment by milot-mirdita** [comment]
Nope, `hhblits_mpi` works on ffindex databases. You need  to leave out the suffix.

**Comment by tamuanand** [comment]
I was able to fix the AVX problem 

Now, I get a different error when I do 
`mpirun -np 256 hhblits_mpi -d <path_to>/uniprot20_2016_02 -i test_fas -oa3m test_a3m_wo_ss -n 3`

Error
`mpirun noticed that process rank 230 with PID 179200 on node  exited on signal 9 (Killed)`
